packages/eslint-plugin-react-hooks/__tests__/ESLintRulesOfHooks-test.js

```diff
@@ -337,6 +337,11 @@ const tests = {
         const [myState, setMyState] = useState(null);
       }
     `,
+    `
+      function _Component() {
```

Comment:
Should we ban this too? I honestly don't know why anyone would use a name like this.

Reply:
I'm not sure why it allows it exactly:
https://github.com/facebook/react/blob/1a2d7925035531e5767ff31ff8d0d581b5f94d49/packages/eslint-plugin-react-hooks/src/RulesOfHooks.js#L42-L46

ddbfe2ed50c7a3476ceff20f5924011ac1ad6428 was the commit but I can't find any context as to why it's allowed.

My assumption was that some people might name internal components like that?

```js
function _InternalComponent() { ... }

export default function Component() {
  return <_InternalComponent />
}
```

Though it seems like a pretty weird naming convention.

Sure thing - I will remove it!